lusidtools/cocoon/dateorcutlabel.py

```diff
@@ -8,14 +8,33 @@
 
 
 class DateOrCutLabel(UserString):
-    def __init__(self, datetime_value):
-        def convert_datetime_utc(datetime_value):
+    def __init__(self, datetime_value, custom_date_format=None):
+        def convert_datetime_utc(datetime_value, custom_date_format=None):
             """
             This function ensures that a variable is a timezone aware UTC datetime
 
             :param any datetime_value:
             :return: datetime datetime_value: The converted timezone aware datetime in the UTC timezone
             """
+            if custom_date_format:
+                if not isinstance(datetime_value, str):
+                    raise TypeError(f"Date {datetime_value} is of type {type(datetime_value)} must be of type 'str' "
+                                    f"when specifying a custom date format. ")
+
+                try:
+                    datetime_value = pd.to_datetime(datetime_value, format=custom_date_format,
```

Comment:
sonar might complain about setting a value to a parameter

Reply:
Why is that a bad thing?